packages/app-project/src/components/AuthModal/components/LoginForm/LoginForm.spec.js

```diff
@@ -1,16 +1,34 @@
 import { shallow } from 'enzyme'
 import React from 'react'
+import sinon from 'sinon'
+import { Formik } from 'formik'
 
 import LoginForm from './LoginForm'
 
 let wrapper
 
 describe('Component > LoginForm', function () {
+  let onSubmitStub = sinon.stub()
   before(function () {
-    wrapper = shallow(<LoginForm />)
+    wrapper = shallow(<LoginForm onSubmit={onSubmitStub} />)
   })
 
   it('should render without crashing', function () {
     expect(wrapper).to.be.ok()
   })
+
+
+  it('should call the `onSubmit` prop on submission', function () {
```

Comment:
If I'm understanding Formik correctly, then I think this test is testing Formik, since the form calls Formik's `handleSubmit` on submission.

`wrapper.find(Formik).simulate('submit', fakeEvent)` should call Formik's `onSubmit` prop, according to the enzyme docs.

Reply:
Isn't it also testing that the component is passing its `onSubmit` prop to the form (regardless of Formik)?

Reply:
That function doesn't get passed to the form. The form calls `handleSubmit`, then Formik calls `onSubmit`.

Reply:
I think it is slightly confusing that the form is decorated with Formik in the view component. Elsewhere, I think we've used a pattern where decorators/HOCs are applied in the container. That threw me a bit, initially, when I was reviewing this.

Reply:
I was trying to follow their docs on how to test: https://github.com/jaredpalmer/formik/blob/f5d76609b222ce583663add279ac76e807e2d0ba/README.md#testing-formik

However, I had trouble applying the example to how we have it setup. Others have been having trouble too: https://github.com/jaredpalmer/formik/issues/937 

And tried out the timeout suggested in one of the comments. I don't think this is ultimately the right way to go.

Reply:
If you name the anonymous functional component that renders the form, I think you could find that by name rather than using `dive()`. It still feels like this test is triggering `handleSubmit` then testing that it calls `onSubmit`, which I'd expect to be tested by Formik anyway.